### ISSUE TITLE
Can't complete release process. Fails at (1) "npm run doc:generate" step (2) websockets dependency / yarn vs npm, (3) more?

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/contracts#276]
**Describe the bug**
I'm trying to do a new release of contracts, because I've improved the README. (And I want to practice before fixing [multi-repo#83](https://github.com/oceanprotocol/multi-repo-issue/issues/83)).

So I'm going through the steps, and it fails at the `npm run doc:generate` step. Wrong compiler version for `solc` or `solidity-docgen`.

And even if I simply skip that, then it fails later on in a websockets dependency.

**To Reproduce**

Steps to reproduce the docgen failure:
1. Go through the steps of the main README, up to the "Releases" section
2. In the "Releases" section, skip the first step "Submit contributions with this workflow." Not needed here, it's an overview.
3. In the "Releases" section, go to second step, the [README for release process](https://github.com/oceanprotocol/contracts/blob/master/docs/RELEASE_PROCESS.md)
4. Go through each step of that README for release process
5. It will fail at the `npm run doc:generate` step. 

Steps to reproduce the websockets dependency failure:
1. Just keep following release process README, and it will fail in the yarn (yarn install) step, complaining about a missing websockets dependency. Note: on travis, it would fail; and I could get it to fail locally in the same way too. I went to try to fix it, but I quickly got tangled up in npm vs yarn and having package.json, package-lock.json, more. (Seems complicated.)

**Expected behavior**

This issue will be considered complete when:
- [ ] It doesn't fail at the issues reported above, and
- [ ] Release process docs have been updated, and other things fixed as needed, so that one can release successfully.

**Additional context**

Versions: 
* ubuntu 18.04
* npm 7.6.3
* node 14.16.0
* solidity-docgen 7.6.3
* solc 7.6.3

Traceback for error (1):
```console
> @oceanprotocol/contracts@0.5.10 doc:generate
> npx solidity-docgen --solc-module solc -o docs/contracts

    Error: Solidity was unable to compile. contracts/DTFactory.sol:1:1: ParserError: Source file requires different compiler 
    version (current compiler is 0.5.17+commit.d19bba13.Emscripten.clang - note that nightly builds are considered to be 
    strictly less than the released version
    pragma solidity 0.5.7;
    ^--------------------^
     (And 9 other errors...)
npm ERR! code 1
npm ERR! path /home/trentmc/code/contracts
npm ERR! command failed
npm ERR! command sh -c npx solidity-docgen --solc-module solc -o docs/contracts

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/trentmc/.npm/_logs/2021-03-18T08_54_28_735Z-debug.log
```

**Comment by trentmc** [comment]
@0x3bfc I'd appreciate your thoughts here. Thank you:)

**Comment by 0x3bfc** [comment]
Will take of it.

**Comment by trentmc** [comment]
> Will take of it.

Thank you @0x3bfc .

**Comment by trentmc** [comment]
👍

**Comment by trentmc** [comment]
I just tried this. It still fails with the same error as reported in the description.

**Comment by trentmc** [comment]
Here's the log mentioned in the last line of the traceback.

[2021-03-20T06_05_25_958Z-debug.log](https://github.com/oceanprotocol/contracts/files/6174986/2021-03-20T06_05_25_958Z-debug.log)

**Comment by 0x3bfc** [comment]
>> I just tried this. It still fails with the same error as reported in the description.



Can you run `npm i` then run `npm run doc:generate` ?

**Comment by trentmc** [comment]
I ran the following.

```console
rm -rf contracts
git clone https://github.com/oceanprotocol/contracts
cd contracts
npm i
npm run doc:generate
```

It worked! The output of the final call was:
```console
> @oceanprotocol/contracts@0.5.10 doc:generate
> npx solidity-docgen --solc-module ./node_modules/solc -o docs/contracts
```

**Comment by trentmc** [comment]
Note: I haven't yet had a chance to see if the next failure is still there.

**Comment by trentmc** [comment]
I went to verify the rest of this, including whether the websockets dependency failure has been fixed. But then I realized: since no updates / fixes have been reported to handle this, I have no reason to expect that it has been fixed.

The appropriate next step for whoever is working on this ticket is: go through the whole release flow to ensure that the process works end-to-end. And fix it as needed.

**Comment by 0x3bfc** [comment]
I think that there is a conflict happens when the user installs the packages using `yarn` then uses `npm` ... So I think this is most probably the reason that some stuff are failing.

I'd recommend to update the issue to `make the release process work through yarn not npm`.

**Comment by trentmc** [comment]
Thanks for the feedback. I changed the issue title & description to highlight that yarn vs npm might be the culprit. (However I did keep it focused on "what" the target outcome is, versus the "how")

**Comment by 0x3bfc** [comment]
solved #283
